src/pages/workspaces/workspace/ShareWorkspaceModal.js

```diff
@@ -27,14 +27,49 @@ const styles = {
   pending: {
     textTransform: 'uppercase', fontWeight: 500,
     color: colors.orange[0]
-  },
-  roleSelect: base => ({
-    ...base,
-    width: 200,
-    marginTop: '0.25rem'
-  })
+  }
 }
 
+const AclInput = ({ value, onChange, disabled }) => {
+  const { accessLevel, canShare, canCompute } = value
+  return div({ style: { display: 'flex', marginTop: '0.25rem' } }, [
+    div({ style: { width: 200 } }, [
+      h(Select, {
+        isSearchable: false,
+        isDisabled: disabled,
+        getOptionLabel: o => Utils.normalizeLabel(o.value),
+        value: accessLevel,
+        onChange: o => onChange({
+          ...value,
+          accessLevel: o.value,
+          ...Utils.switchCase(o.value,
+            ['READER', () => ({ canCompute: false })],
+            ['WRITER', () => ({ canCompute: true })],
+            ['OWNER', () => ({ canCompute: true, canShare: true })]
```

Comment:
I noticed that when you switch from "Owner" to "Writer", the canShare and canCompute checks are auto-checked. When you go from "Reader" to "Writer", though, only the "canCompute" check is auto-checked.

Reply:
This follows the FC behavior, which I'm inclined to leave for now.